### PR TITLE
[RNMobile] Add url option to image block

### PR DESCRIPTION
**Pull request by derekblank**
This is a PR for running CI checks on [gutenberg#40334](https://github.com/WordPress/gutenberg/pull/40334).

To test: N/A

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/RELEASE-NOTES.txt) if necessary.

### CODE REVIEW

**Comment by derekblank** [comment]
@fluiddot I believe this is the correct way to keep the Gutenberg ref updated for [gutenberg#40334](https://github.com/WordPress/gutenberg/pull/40334). Let me know if I've missed something for the matching Gutenberg Mobile PR here. 🙇

File: .gitmodules [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [submodule "gutenberg"]
 	path = gutenberg
-	url = ../../WordPress/gutenberg.git
+	url = https://github.com/derekblank/gutenberg.git
```
Comment (fluiddot) [inline .gitmodules]: The URL of the Gutenberg submodule should remain pointing to its original value `../../WordPress/gutenberg.git`.

**Review by fluiddot — CHANGES_REQUESTED** [review]
> @fluiddot I believe this is the correct way to keep the Gutenberg ref updated for https://github.com/WordPress/gutenberg/pull/40334. Let me know if I've missed something for the matching Gutenberg Mobile PR here. 🙇

@derekblank Yeah, this is the correct way although I spotted a couple of issues that we'd need to check:
- We shouldn't modify the `.gitmodules`.
- The Gutenberg reference should point to the latest commit, now it's pointing to `ee61ccb3b1eabce9dbc32fbd6c47cc64277002e7` but the last commit from the [Gutenberg PR](https://github.com/WordPress/gutenberg/pull/40334) is `a43fce12c6bd88c4b2307283186a1a54a3b2967b`.

Additionally, it would be great to add an entry to the changelog files as outlined in the PR's submission checklist:
> I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/RELEASE-NOTES.txt) if necessary.

On a different note, you can also generate this PR automatically by adding the `Mobile app - Automation` label to the Gutenberg PR ([here](https://github.com/wordpress-mobile/gutenberg-mobile/pull/4857) you can check an example).

**Review by fluiddot — APPROVED** [review]
Approved via https://github.com/WordPress/gutenberg/pull/40334#pullrequestreview-980017692.

**NOTE:** Before merging the PR, the Gutenberg reference should point to the merge commit in `trunk` after merging the [Gutenberg PR](https://github.com/WordPress/gutenberg/pull/40334).

**Comment by derekblank** [comment]
@fluiddot Thanks for the tips! The automated PR would've been much easier. :) I think I had my wires crossed for a second, but I think I've sorted it out now after updating the changelog and the gutenberg refs. Since this PR is not automated and I cannot merge [gutenberg#40334](https://github.com/WordPress/gutenberg/pull/40334) myself, I think the next step for this PR is just to keep the submodule updated until the gutenberg PR can be merged?

**Comment by fluiddot** [comment]
> I think I had my wires crossed for a second, but I think I've sorted it out now after updating the changelog and the gutenberg refs. Since this PR is not automated and I cannot merge https://github.com/WordPress/gutenberg/pull/40334 myself, I think the next step for this PR is just to keep the submodule updated until the gutenberg PR can be merged?

@derekblank Yeah exactly, the next step for this PR before merging it is to update the Gutenberg reference once https://github.com/WordPress/gutenberg/pull/40334 is merged.

Since you don't have permission to merge, I can take over that 👍. In relation to this, I'll restart the failing jobs in the Gutenberg PR (looks like they fail due to flakiness) and once everything is 🟢  I'll merge it. I'll let you know when it's merged so you can update the Gutenberg reference here.

**Comment by fluiddot** [comment]
@derekblank I've just merged the [Gutenberg PR](https://github.com/WordPress/gutenberg/pull/40334), here is the commit SHA: [`20f6ca5806b753593496d2c7bf80f5bd643b34a1`](https://github.com/WordPress/gutenberg/commit/20f6ca5806b753593496d2c7bf80f5bd643b34a1).

File: RELEASE-NOTES.txt [hunk 2]
```diff
@@ -1,6 +1,7 @@
 Unreleased
 ---
 * [*] [a11y] Improve text read by screen readers for BottomSheetSelectControl [https://github.com/wordpress-mobile/gutenberg-mobile/pull/4854]
+* [*] [Image block] Add Insert from URL as a media option
```
Comment (fluiddot) [inline RELEASE-NOTES.txt]: We usually add a reference to the Gutenberg Mobile PR along with the entry, e.g.:
```
[a11y] Improve text read by screen readers for BottomSheetSelectControl [https://github.com/wordpress-mobile/gutenberg-mobile/pull/4854]
```
Could you also add it for this change? Thanks 🙇 !

**Comment by derekblank** [comment]
@fluiddot Thanks for merging the Gutenberg PR. I've updated the release notes, and set the gutenberg reference to [`20f6ca5806b753593496d2c7bf80f5bd643b34a1`](https://github.com/WordPress/gutenberg/commit/20f6ca5806b753593496d2c7bf80f5bd643b34a1). 

For non-automated gutenberg-mobile PRs (like this one), when the associated gutenberg PR has been merged, would the author (i.e., me) then merge the gutenberg-mobile PR? Or is there another approval step needed? I see all of the checks have passed but I just want to confirm that it is ok to merge 🙇 .

**Comment by fluiddot** [comment]
> @fluiddot Thanks for merging the Gutenberg PR. I've updated the release notes, and set the gutenberg reference to [`20f6ca5806b753593496d2c7bf80f5bd643b34a1`](https://github.com/WordPress/gutenberg/commit/20f6ca5806b753593496d2c7bf80f5bd643b34a1).

🎊 
 
> For non-automated gutenberg-mobile PRs (like this one), when the associated gutenberg PR has been merged, would the author (i.e., me) then merge the gutenberg-mobile PR? Or is there another approval step needed? I see all of the checks have passed but I just want to confirm that it is ok to merge 🙇 .

Yeah, once both Gutenberg and Gutenberg-Mobile PRs are approved and ready to merge, usually the author is in charge of updating the references and merging the PRs. In this case, since everything is ready, including the PR checks, you can go ahead and merge the PR :shipit:.

This also applies to the automated Gutenberg-Mobile PRs, as only updating the Gutenberg reference is the part automated. Update the reference to point to the merge commit and PR merging is still manual.

**Pull request merged**